dask/layers.py

```diff
@@ -43,16 +50,19 @@ def __call__(self, *args, **kwargs):
 #
 
 
-class CreateArrayDeps(BlockwiseDep):
-    """Index-chunk mapping for BlockwiseCreateArray"""
+class ArrayBlockwiseDep(BlockwiseDep):
+    """
+    Blockwise dep for array-likes, which only needs chunking
+    information to compute its data.
+    """
```

Comment:
```suggestion
    """
    chunks: tuple[tuple[int, ...], ...]
    numblocks: tuple[int, ...]
    produces_tasks: bool = False
```
I like having type annotations at the class level, for better error-catching and for documentation purposes

Reply:
I suppose so... I don't like this flavor of annotation very much largely because I find the [`ClassVar`](https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations) choice to be baffling. But some annotation is better than none.